batch2/batch/globals.py

```diff
@@ -3,8 +3,9 @@
 complete_states = ('Cancelled', 'Error', 'Failed', 'Success')
 
 valid_state_transitions = {
-    'Pending': {'Running'},
-    'Running': {'Running', 'Cancelled', 'Error', 'Failed', 'Success'},
+    'Pending': {'Ready'},
```

Comment:
You can't go from Pending to Cancelled?

Reply:
No,  You go from pending => ready when your parents complete in the mark_job_complete stored procedure.  You go from ready => cancelled in the scheduler.

Note, this isn't actually checked anymore since the transitions all happen in the database now.